pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,59 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// DiskType is used to specify the disk type for a logical disk, e.g. hdd or ssd.
+type DiskType string
+
+// Valid disk types for RAID configuration
+const (
+	HDD DiskType = "hdd"
+	SSD DiskType = "ssd"
+)
+
+// RAIDLevel type is used to specify the RAID level for a logical disk.
+type RAIDLevel string
+
+// Valid RAID levels
+const (
+	RAID0  RAIDLevel = "0"
+	RAID1  RAIDLevel = "1"
+	RAID10 RAIDLevel = "1+0"
+)
+
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.  If unspecified, the maximum capacity of disk will be used for logical disk.
+	// +kubebuilder:validation:Minimum=1
+	SizeGb *int `json:"sizeGb,omitempty"`
```

Comment:
Is anyone ever likely to ask for a 0 byte volume? Could we use a regular int instead of a pointer, and treat a value of 0 as meaning that we should use the maximum space available like we do with NumberOfPhysicalDisks?